velero-plugins/configmap/backup.go

```diff
@@ -0,0 +1,78 @@
+package configmap
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/konveyor/openshift-velero-plugin/velero-plugins/common"
+	"github.com/konveyor/openshift-velero-plugin/velero-plugins/util/openshift"
+	"github.com/sirupsen/logrus"
+	v1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// BackupPlugin is a backup item action plugin for Velero.
+type BackupPlugin struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo returns a velero.ResourceSelector that applies to configmaps.
+func (p *BackupPlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"configmaps"},
+	}, nil
+}
+
+// Execute check if configMap should be skipped restore due to
```

Comment:
... skipped restore due...

backup instead of restore, or just remove restore word?

Reply:
We are not skipping backup here. We aim to find out which of these items we will backup anyway but not restore.

Reply:
We will annotate such objects that will be in velero backup tarball should it ever be needed but will programmatically skip restores of these configmaps.

Reply:
This work has to be done at backup time while we are still able to query for pod that is in ownerRef. At restore time if disaster already happened it is not possible to check if the pod referenced is a build pod or not.

Reply:
right, understood

would rewrite to be more clear, something like

> Execute add annotation to skip restore of configMap belonging to a build pod, which will get regenerated on build restore, to avoid restoring cacert from old cluster into new environment